### PR TITLE
test(api): verify that null isin behavior is consistent across backends

### LINKED ISSUE #8079: bug: x.isin(<contains NULLs>) result depends on backend
### What happened?

```python
t1 = ibis.memtable({"x": [1, 2]})
e = t1.x.isin([1, None])
print(ibis.duckdb.connect().execute(e))
print(ibis.sqlite.connect().execute(e))
print(ibis.polars.connect().execute(e))
print(ibis.pandas.connect().execute(e))
```
yields

0    True
1    None
Name: InValues(x), dtype: object
0    True
1    None
Name: InValues(x), dtype: object
0     True
1    False
Name: InValues(x), dtype: bool
0    True
1    None
Name: InValues(x), dtype: object

If a needle isn't present in the haystack, and the haystack contains NULL, then
for some backends this returns False, and for others NULL.

I think it is pretty standard SQL to return null. But this isn't what I expected to happen, I actually expected
pandas semantics. IDK which we want, but they should be the same, and this should be documented.

### What version of ibis are you using?

main

### What backend(s) are you using, if any?

varies

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by cpcloud**
Closes #8079.

### CODE REVIEW

File: ibis/expr/types/generic.py [hunk 1]
```diff
@@ -498,6 +498,9 @@ def between(
     def isin(self, values: Value | Sequence[Value]) -> ir.BooleanValue:
         """Check whether this expression's values are in `values`.
 
+        If any of `values` is `NULL` or a row of `self` is `NULL` the result
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/expr/types/generic.py:501]: Are taking issue with the phrasing, or do you think what I wrote is incorrect?

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/expr/types/generic.py:501]: It's a bit moot until I can iron out the implementations :)

[the remark this answers is not in the document]
Comment (NickCrews) [reply ibis/expr/types/generic.py:501]: The "If any of `values` is `NULL`" isn't quite complete, the other part of the condition is that `self` isn't in `values`.

[the remark this answers is not in the document]
Comment (NickCrews) [reply ibis/expr/types/generic.py:501]: eg the current phrasing implies that `4 in [4, NULL]` will give NULL, but it actually gives True

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/expr/types/generic.py]: Fixed.

**Comment by cpcloud** [comment]
There's no obvious way to support this behavior in the MS SQL backend since it makes using boolean operations outrageously difficult.

All boolean operations must be wrapped in `IIF` because MS SQL lacks a boolean type (but **requires** what it calls boolean expressions in certain places, it's extremely difficult to work with).

The follow is not allowed:

```sql
SELECT IIF(IIF(x IN (NULL), NULL, NULL), 1, 0)
```

because the output of the inner `IIF` is `NULL` which isn't a valid boolean expression.

**Review by kszucs — APPROVED** [review]

**Pull request merged**